engine/storage/volume/src/main/java/org/apache/cloudstack/storage/volume/VolumeServiceImpl.java

```diff
@@ -2794,6 +2799,16 @@ public void checkAndRepairVolumeBasedOnConfig(DataObject dataObject, Host host)
         }
     }
 
+    @Override
+    public void validateChangeDiskOfferingEncryptionType(long existingDiskOfferingId, long newDiskOfferingId) {
+        DiskOfferingVO existingDiskOffering = diskOfferingDao.findById(existingDiskOfferingId);
+        DiskOfferingVO newDiskOffering = diskOfferingDao.findById(newDiskOfferingId);
```

Comment:
in any case, new or old offering is removed, checked before this call?

Reply:
Added findbyincludingremoved() method on existing disk offering.